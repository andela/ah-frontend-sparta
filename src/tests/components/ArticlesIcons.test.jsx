import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ArticlesIcons from '../../components/Articles/articlesIcons';

describe('Article Icons  Component', () => {
  let mountApp;
  const props = {
    article: {
      id: 1,
      author: {
        username: 'rogha',
      },
      article_read_time: 'less than a minute read',
      title: 'Post one',
      description: 'too long a story',
      slug: 'post-one-first',
      body: 'ahhhhh --edit me',
      createdAt: '2019-04-18T11:55:58.111843Z',
      tags: ['Arsenal', 'soccer', 'Andela'],
    },
  };
  beforeEach(() => {
    mountApp = renderer.create(
      <Router>
        <ArticlesIcons {...props} />
      </Router>,
    );
  });

  it('Should render without crashing', () => {
    expect(mountApp).toMatchSnapshot();
  });
});
