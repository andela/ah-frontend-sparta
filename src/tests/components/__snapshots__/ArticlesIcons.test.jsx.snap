// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Icons  Component Should render without crashing 1`] = `
<div
  className="icons"
>
  <div
    className="btn-group dropleft"
  >
    <a
      aria-expanded="false"
      aria-haspopup="true"
      data-toggle="dropdown"
      href="sharing"
    >
      <i
        className="fas fa-share-alt-square"
      />
    </a>
    <div
      className="dropdown-menu"
    >
      <a
        className="dropdown-item"
        href="https://www.facebook.com/sharer/sharer.php?u=undefined/articles/post-one-first"
      >
        <i
          className="fab fa-facebook-square facebook-icon"
        />
        Facebook
      </a>
      <a
        className="dropdown-item"
        href="https://twitter.com/intent/tweet?url=undefined/articles/post-one-first&text=too long a story"
      >
        <i
          className="fab fa-twitter twitter-icon"
        />
        Twitter
      </a>
      <a
        className="dropdown-item"
        href="mailto:?Subject=Post%20one&body=I found this article to be worth reading, Post one undefined/articles/post-one-first"
      >
        <i
          className="fas fa-envelope-square envelop-icon"
        />
        Mail
      </a>
    </div>
  </div>
     
  <a
    href="comment"
  >
    <i
      className="far fa-comment"
    />
  </a>
     
  <a
    href="like"
  >
    <i
      className="far fa-thumbs-up pull float-right"
    />
  </a>
     
  <a
    href="dislike"
  >
    <i
      className="far fa-thumbs-down"
    />
  </a>
     
</div>
`;
